src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -26,10 +26,11 @@ public static partial class JsonSerializer
             Debug.Assert(state.Current.ReturnValue != default || state.Current.TempDictionaryValues != default);
             Debug.Assert(state.Current.JsonClassInfo != default);
 
-            if (state.Current.IsProcessingDictionaryOrIDictionaryConstructible() &&
+            bool isProcessingDictObject = state.Current.IsProcessingDictionaryOrIDictionaryConstructibleObject();
```

Comment:
Is this the only place where such caching of redundant conditions would be useful? I thought there were others.

Reply:
I didn't find another. The remaining redundant checks were removed with the bit flags.